### PR TITLE
Add the capability to store and interact with properties needed for displaying onboarding screens.

### LINKED ISSUE #97: Add `isOwnerLoggedOnce` and `isSitterLoggedOnce` property to app preferences.
## This task is a part of the issue:
 - #96 

## Description
Add a new property to the owner's or sitter's settings to track whether they are using the application for the first time.

### PR DESCRIPTION
**Pull request by iBYarchitect**
## Summary
This PR is a part of the work to resolve #97 

## Changes
- Added `showOwnerOnboarding` and `showSitterOnboarding` properties to struct`Preferences`
- Added tests for methods `set(showOwnerOnboarding: Bool)`, `set(showSitterOnboarding: Bool)`, `getShowOwnerOnboarding()`, `getShowSitterOnboarding()`

## Checks to complete
- [x] The PR title has a conventional commit style (short and descriptive).
- [x] The PR description includes essential information that helps understand the purpose of the changes.
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] I've updated `CHANGELOG.md` file.
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -10,6 +10,7 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.1.0/).
 
 - Add `Walking` entity - [#88](https://github.com/IronFoundation/iOS-Woof/pull/88)
 - Add `City` property to the `owner` entity - [#93](https://github.com/IronFoundation/iOS-Woof/pull/93)
+- Add `isOwnerLoggedOnce` and `isSitterLoggedOnce` properties to struct`Preferences` - [#100](https://github.com/IronFoundation/iOS-Woof/pull/100)
```
Comment (NadzeyaShpakouskaya) [inline CHANGELOG.md:13]: We added not only properties but methods as well. Do you think we need to mention that? Maybe we can express that in a more general way?
```suggestion
- Add the possibility to track if the owner/sitter was logged in at least once  - [#100](https://github.com/IronFoundation/iOS-Woof/pull/100)
```

File: Woof/WoofTests/Model/Preferences/PreferencesHandlerSetOwnerLoggedTests.swift [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+final class PreferencesHandlerSetOwnerLoggedTests: XCTestCase {
+    private var userPreferencesStorage = KeyValueStorage.testPreferencesStorage
+
+    override func setUp() {
+        super.setUp()
+        userPreferencesStorage.deleteData(for: KeyValueStorage.Key.userPreferences)
+    }
+
+    func testExistsInAPI() {
+        _ = PreferencesHandler.setOwnerLoggedOnce(true)
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/Model/Preferences/PreferencesHandlerSetOwnerLoggedTests.swift:12]: nit
```suggestion
    func testMethodExistsInAPI() {
        _ = PreferencesHandler.setOwnerLoggedOnce(true)
```
You can keep your version or review the rest of code

File: Woof/WoofTests/Model/Preferences/PreferencesHandlerSetOwnerLoggedTests.swift [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+    }
+
+    func testOverridesThePreviouslySavedOwnerLoggedOnceValue() {
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/Model/Preferences/PreferencesHandlerSetOwnerLoggedTests.swift:15]: nit
```suggestion

    func testMethodSuccesfullyOverridesThePreviouslySavedValue() {
```

File: Woof/WoofTests/Model/Preferences/PreferencesTests.swift [hunk 4]
```diff
@@ -24,4 +24,42 @@ final class PreferencesTests: XCTestCase {
 
         XCTAssertEqual(preferences.selectedRole, Role.owner)
     }
+
+    func testPreferencesReturnFalseForIsOwnerLoggedOnceByDefault() {
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/Model/Preferences/PreferencesTests.swift:28]: nit
```suggestion
    func testPreferencesInitializedWithExpectedDefaultValueForIsOwnerLoggedOnce() {
```

**Review by NadzeyaShpakouskaya — COMMENTED** [review]
Nice attempt 🙃 
I want to clarify some points:
 - the naming of properties `isOwnerLoggedOnce` and `isSitterLoggedOnce` is confusing to me.
If we use them to present the onboarding flow (filling in the basic information is a part of an onboarding process I guess), maybe it makes sense to call it more specific.
- why do we need to create methods to change properties if we can change properties directly in the preferences. It would make sense if the property were `private(set)`, and we couldn't change them directly.

File: Woof/Woof/Model/Preferences/PreferencesHandler.swift [hunk 5]
```diff
@@ -23,6 +23,38 @@ enum PreferencesHandler {
         return store(preferences)
     }
 
+    /**
+     Saves the information whether the owner has logged in at least once.
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/Model/Preferences/PreferencesHandler.swift]: Fixed.

File: CHANGELOG.md (hunk 1 above)
Comment (iBYarchitect) [reply CHANGELOG.md:13]: Fixed

File: Woof/WoofTests/Model/Preferences/PreferencesHandlerSetOwnerLoggedTests.swift (hunk 2 above)
Comment (iBYarchitect) [reply Woof/WoofTests/Model/Preferences/PreferencesHandlerSetOwnerLoggedTests.swift]: Thx fixed with 423791b

File: Woof/WoofTests/Model/Preferences/PreferencesHandlerSetOwnerLoggedTests.swift (hunk 3 above)
Comment (iBYarchitect) [reply Woof/WoofTests/Model/Preferences/PreferencesHandlerSetOwnerLoggedTests.swift]: fixed with 47530db

File: Woof/WoofTests/Model/Preferences/PreferencesTests.swift (hunk 4 above)
Comment (iBYarchitect) [reply Woof/WoofTests/Model/Preferences/PreferencesTests.swift]: fixed with 47530db

**Review by NadzeyaShpakouskaya — APPROVED** [review]
LGTM

**Pull request merged**